posts/fantastic-simprocs.md

```diff
@@ -0,0 +1,218 @@
+---
+author: 'Yaël Dillies, Paul Lezeau'
+category: 'Metaprogramming'
+date: 2025-04-16 12:00:00 UTC+01:00
+description: 'Informal introduction to simprocs and what they are useful for'
+has_math: true
+link: ''
+slug: fantastic-simprocs
+tags: 'simp, simproc, meta'
+title: Fantastic Simprocs and Where to Find Them.
+type: text
+---
+
+Lean core added support for custom simplification procedures, aka *simprocs*.
+This blog post is the first in a series of two aimed at explaining what a simproc is, what kind of problems can be solved with simprocs, and what tools we have to write them.
```

Comment:
Style question; should we write `` `simproc` `` everywhere?

Reply:
... or `Simproc`? I have been following the convention that formal concepts are quoted, while informal ones aren't. In this blog post, a simproc is an informal concept (but `simp` is a formal one), and simprocs become the formal `Simproc`s in the followup blog post

Reply:
Both `Lean.Meta.Simp.Simproc` and `simproc` are valid pieces of lean syntax, so I think the latter is fine to use instead of the former.

My heuristic is "if it's not a real word then always write and quote it as code", but it's not my blog post so I'll let you make the final call.

Reply:
For comparison, the reference manual also refers to simproc without code indicators, so this is at least consistent with that

Reply:
Can we link the reference manual from the blog post?

Reply:
Sure, I think it makes sense to mention it at the end of the "how simp works" section, saying something like, you can read about it in more detail here: https://lean-lang.org/doc/reference/latest////The-Simplifier/#the-simplifier

Reply:
There's not much about simprocs there, only a paragraph here https://lean-lang.org/doc/reference/latest////The-Simplifier/Rewrite-Rules/#simp-rewrites

Reply:
You could also link https://leanprover-community.github.io/extras/simp.html#introduction

Reply:
My vote is to italicize it on first use but otherwise treat it like a regular word (which seems to be already happening).